lsp/nls/src/requests/completion.rs

```diff
@@ -80,22 +109,37 @@ lazy_static! {
     // This regexp must be the reverse of the regexp in the lexer (nickel_lang::parser::lexer)
     // to correctly parse identifiers.
     static ref RE: regex::Regex = regex::Regex::new(r"[_a-zA-Z0-9-']*[a-zA-Z]_?").unwrap();
+    static ref RE_SPACE: regex::Regex = regex::Regex::new(r"\s+").unwrap();
 }
 
-/// Get the identifier before the `.`, for record completion.
-fn get_identifier_before_dot(text: &str) -> Option<String> {
+/// Get the string chunks that make up an identifier path.
+fn get_identifier_path(text: &str) -> Option<Vec<String>> {
+    // skip any `.` at the end of the text
     let text: String = text
         .chars()
         .rev()
         .skip_while(|c| *c == '.') // skip . (if any)
+        .collect::<String>()
+        .chars()
+        .rev()
         .collect();
 
-    let name = RE.find(&text)?;
-    Some(name.as_str().chars().rev().collect())
+    let result: Vec<_> = RE_SPACE.split(text.as_ref()).map(String::from).collect();
+    if result.is_empty() {
+        return None;
+    }
```

Comment:
You can remove these lines, since building an iterator on an empty vec is essentially a no-op, and `get` (or `next` if you follow my suggestion) will return `None`, which means the `?` will result in the whole function returning `None`.